packages/react-components/react-jsx-runtime/src/jsx-runtime.test.tsx

```diff
@@ -266,20 +266,29 @@ describe('createElement with assertSlots', () => {
     });
 
     it('keeps children from a render template in a render callback', () => {
-      type TestComponentSlots = { outer: NonNullable<Slot<'div'>>; inner: NonNullable<Slot<'div'>> };
+      const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {
+        /* noop */
+      });
+      type TestComponentSlots = {
+        outer: NonNullable<Slot<'div'>>;
+        inner1: NonNullable<Slot<'div'>>;
+        inner2: NonNullable<Slot<'div'>>;
+      };
       type TestComponentState = ComponentState<TestComponentSlots>;
       type TestComponentProps = ComponentProps<Partial<TestComponentSlots>>;
 
       const TestComponent = (props: TestComponentProps) => {
         const state: TestComponentState = {
-          components: { outer: 'div', inner: 'div' },
-          inner: slot.always(props.inner, { defaultProps: { id: 'inner' }, elementType: 'div' }),
+          components: { outer: 'div', inner1: 'div', inner2: 'div' },
+          inner1: slot.always(props.inner1, { defaultProps: { id: 'inner-1' }, elementType: 'div' }),
+          inner2: slot.always(props.inner2, { defaultProps: { id: 'inner-2' }, elementType: 'div' }),
```

Comment:
do you actually need a second slot here? since the outer slot is rendered with a dynamic element it should always be rendered as a dynamic element

Reply:
Yes. What dictates if an element is rendered with `jsxs` instead of `jsx` is having a fixed amount of children bigger than one, so to force the render of `outer` slot as `jsxs` it need inside of it at least 2 well defined children.